qiskit/optimization/algorithms/optimization_algorithm.py

```diff
@@ -140,8 +156,8 @@ class OptimizationResult:
 
     def __init__(self, x: Union[List[float], np.ndarray], fval: float,
                  variables: List[Variable],
-                 raw_results: Optional[Any] = None,
-                 status: OptimizationResultStatus = OptimizationResultStatus.SUCCESS) -> None:
+                 status: OptimizationResultStatus,
```

Comment:
So the order of these was switched since status was required no longer to have a default I guess. In the future we have to be careful over these types of breaking changes since other external code can implement an optimizer and extend the result and now potentially be broken though such a change.

Reply:
You are right. Yes, this is dangerous, but this is a drawback of having default values. By doing this I want to enforce more conscious logic around status. I hope it won't break anything.